### ISSUE TITLE
Variance annotation on Angstrom.t

### ISSUE THREAD

**Issue by andreas** [inhabitedtype/angstrom#61]
Currently there's no variance annotation on `Angstrom.t`, which is unfortunate when using polymorphic variants: I cannot have a value of type ```[`Foo] Angstrom.t``` that I can reuse for ```[`Foo | `Bar] Angstrom.t``` and ```[`Foo | `Baz] Angstrom.t```.

Example:

```ocaml
let x : [`Foo] Angstrom.t  = Angstrom.return `Foo;;
let y = (x :> [`Foo | `Bar] Angstrom.t);;
Error: Type [ `Foo ] Angstrom.t is not a subtype of [ `Bar | `Foo ] Angstrom.t                                                                                                              The first variant type does not allow tag(s) `Bar
```

If `'a Angstrom.t ` is declared as covariant in the signature, then the above example works. I've made this modification in a local copy of the repo and everything seems to work. I'm not sure whether there are any negative implications though (besides constraining how you can use `'a` in the implementation). What are your thoughts on making `'a Angstrom.t` covariant?

**Comment by seliopou** [comment]
This was an oversight as opposed to an intentional design decision. I don't think there should be any negative implications. Send a PR over and I'll take a look.

-Spiros E.

i sent this from my Phone.

**Comment by seliopou** [comment]
Closed by #62.
